packages/pages/src/components/image/image.tsx

```diff
@@ -0,0 +1,210 @@
+import * as React from "react";
+import { useEffect, useRef, useState } from "react";
+import { ImageProps, ImageLayout } from "./types";
+
+const MKTGCDN_URL_REGEX =
+  /((?<=^http:\/\/a\.mktgcdn\.com\/p\/)|(?<=^https:\/\/a\.mktgcdn\.com\/p\/)).+(?=\/(.*)$)/g;
+enum imgLoadingStatus {
+  NONE = "none",
+  LOADED = "loaded",
+  ERROR = "error",
+}
+
+/**
+ * Renders an image based from the Yext Knowledge Graph. Example of using the component to render
+ * simple and complex image fields from Yext Knowledge Graph:
+ * ```
+ * import { Image } from "@yext/pages/components";
+ *
+ * const simpleImage = (<Image image={document.logo} />);
+ * const complexImage = (<Image image={document.photoGallery[0]} />);
+ * ```
+ *
+ * @public
+ */
+export const Image = ({
+  image,
+  className,
+  width,
+  height,
+  aspectRatio,
+  layout = ImageLayout.INTRINSIC,
+  placeholder,
+  imgOverrides,
+  style = {},
+}: ImageProps) => {
+  const imgRef = useRef<HTMLImageElement>(null);
+  const [imgStatus, setImgStatus] = useState<imgLoadingStatus>(
+    imgLoadingStatus.NONE
+  );
+
+  useEffect(() => {
+    if (imgRef.current?.complete) {
+      setImgStatus(imgLoadingStatus.LOADED);
+    }
+  }, []);
+
+  if (layout != ImageLayout.FIXED && (width || height)) {
+    console.warn(
+      "Width or height is passed in but layout is not fixed. These will have no impact. If you want to have a fixed height or width then set layout to fixed."
+    );
+  }
+
+  const imgWidth: number = image.image.width;
+  const imgHeight: number = image.image.height;
+  const imgUUID = getImageUUID(image.image.url);
+
+  const { src, imgStyle, widths } = handleLayout(
+    layout,
+    imgWidth,
+    imgHeight,
+    imgUUID,
+    style,
+    width,
+    height,
+    aspectRatio
+  );
+
+  // Generate Image Sourceset
+  const srcSet: string = widths
+    .map((w) => `${getImageUrl(imgUUID, w, (imgHeight / imgWidth) * w)} ${w}w`)
+    .join(", ");
+
+  const onError = () => {
+    setImgStatus(imgLoadingStatus.ERROR);
+    console.warn(`Invalid image src: ${src}.`);
+  };
+
+  return (
+    <>
+      {placeholder != null &&
```

Comment:
Just to be certain there's never an instance in which both placeholder and <img> could render I'd recommend just switching on imgLoadingStatus.LOADED. If it's loaded, <img>, otherwise return placeholder. I don't think it'll be an issue if it's null, right? That's equivalent to what'd happen if we were in an error state w/o a placeholder anyway I think

Reply:
Not sure if I get your point. But the main issue with this is, we want to make sure <img> is rendered in order to trigger onLoad, which sets the imgStatus. I verified that the <img> and placeholder never show up together

Reply:
Jim, I think the issue here is that the placeholder might not exist. We only display it if there is one.